### PR TITLE
Money transfers

### PR DESCRIPTION
**Pull request by AlexMxwl**
This branch implements four related cards from Trello:
1. As a user, I would like to be able to transfer money between my accounts
2. As a user, I would like to be able to deposit money to my account
3. As a user, I would like to be able to pay another user
4. PAY returns new account balance alongside Success/Failed

Three new user functions have been added:
- DEPOSIT
- MOVE
- PAY

In order to test these functions, I gave the other customer accounts passwords so it is possible to log in to them and confirm that money has moved between accounts. All other changes should be directly related to the three new functions above.

There is one bug that I am aware of - if you enter an invalid username for someone that you are trying to PAY, the client will print "null" in an infinite loop - I suspect that this is the same issue that we get when logging in with invalid credentials, and it is something that should be solved on a separate branch.

### CODE REVIEW

File: newbank/server/Account.java [hunk 1]
```diff
@@ -3,15 +3,31 @@
 public class Account {
 	
 	private String accountName;
-	private double openingBalance;
+	private double accountBalance;
```
Comment (adrianlio) [inline newbank/server/Account.java:6]: Good thinking here!

File: newbank/server/NewBank.java [hunk 2]
```diff
@@ -91,7 +100,137 @@ private String showHelp() {
...
+				if(depositAmount <= 0) {
+					// a deposit amount must be positive
+					inputErrorPrompts += "Deposit amount '" + requestParams[1] + "' is not valid.\n";
+				}
+			} catch (NumberFormatException e) {
+				inputErrorPrompts += "Deposit amount '" + requestParams[1] + "' is not valid.\n";
+			}
+			Account depositAccount = customers.get(customer.getKey()).getAccount(requestParams[2]);
+			if(depositAccount == null) {
+				inputErrorPrompts += "Account for withdrawal '" + requestParams[2] + "' does not exist.\n";
```
Comment (adrianlio) [inline newbank/server/NewBank.java:130]: If we are depositing money, should the error prompt be "Account for deposit" instead?

File: newbank/server/NewBank.java [hunk 3]
```diff
@@ -91,7 +100,137 @@ private String showHelp() {
...
+			if(inputErrorPrompts.length() > 0) {
+				return "Transfer could not be made: \n" + inputErrorPrompts;
+			}
+			// transfer funds between the specified accounts
+			withdrawalAccount.withdrawFunds(transferAmount);
+			depositAccount.depositFunds(transferAmount);
+			return transferAmount + " transferred from " + withdrawalAccount.getName() + " to " +
+					depositAccount.getName();
+		}
+		return "Invalid entry. Try TRANSFERFUNDS <amount> <account to withraw from> <account to deposit to>";
```
Comment (adrianlio) [inline newbank/server/NewBank.java:184]: Minor cosmetic change - should TRANSFERFUNDS be renamed to MOVE?

**Review by adrianlio — CHANGES_REQUESTED** [review]
Looks good! Tested the functions and all works as expected but note the issue around the getting the Payee and the 'null' infinite loop which we will need to address in the sprint backlog as you said. Otherwise, just some minor cosmetic changes but otherwise, super nice!

File: newbank/server/NewBank.java (hunk 2 above)
Comment (AlexMxwl) [reply newbank/server/NewBank.java:130]: Yes, good spot - clearly I was lazy in my copy + pasting!

File: newbank/server/NewBank.java (hunk 3 above)
Comment (AlexMxwl) [reply newbank/server/NewBank.java:184]: Yep, agreed - I changed the name of the command last minute but forgot this line!

**Review by adrianlio — APPROVED** [review]
Great, thanks for that!

**Pull request merged**